File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/dto/GetListedStockPastFinancialStatementsResponse.java
package com.kbank.convenience.stock.stock.mediation.client.dto;

import lombok.Builder;

import java.time.LocalDate;
import java.util.List;

@Builder
public record GetListedStockPastFinancialStatementsResponse(
        String itemCodeNumber,
        String targetStatement,
        List<PastFinancialStatement> list

) {
    @Builder
    public record PastFinancialStatement(
            LocalDate baseDate,
            Long amount
    ) {}
}

File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/dto/GetListedStockPastFinancialStatementsResponse.kt
package com.kbank.convenience.stock.stock.mediation.client.dto

import lombok.Builder
import java.time.LocalDate

@Builder
@JvmRecord
data class GetListedStockPastFinancialStatementsResponse(val itemCodeNumber: String,
                                                         val targetStatement: String,
                                                         val list: List<PastFinancialStatement>

) {
    @Builder
    @JvmRecord
    data class PastFinancialStatement(
                                    val baseDate: LocalDate,
                                    val amount: Long
    )
}
